src/rules/selector-class-pattern/index.js

```diff
@@ -13,23 +14,46 @@ export const messages = ruleMessages(ruleName, {
   expected: selectorValue => `Expected class selector ".${selectorValue}" to match specified pattern`,
 })
 
-export default function (pattern) {
+export default function (pattern, options) {
   return (root, result) => {
     const validOptions = validateOptions(result, ruleName, {
       actual: pattern,
       possible: [ isRegExp, isString ],
+    }, {
+      actual: options,
+      possible: {
+        resolveNestedSelectors: isBoolean,
+      },
+      optional: true,
     })
     if (!validOptions) { return }
 
+    // In context of a nested selector, some combinations
+    // are tested mutliple times, to avoid triggering multiple
+    // identical errors, we will use an internal list of checked rules
+    var checked = []
```

Comment:
Oh I see. I forgot about how this rule only checks the simple class selector, not the full maybe -compound-complex selector. So with that in mind am I right that the resolution of nested selectors only matters when there's a `&` that is *part of* a simple selector (e.g. this doesn't matter `& .bar` or `& > .bar` but these do: `&bar`, `bar&`). 

If that's the case, maybe the rule could be made more efficient by only using `resolveNestedSelectors` if `&` is found and is *part of a simple selector* — instead of resolving *all* the selectors. Does that make sense?